pkg/guest/guest.go

```diff
@@ -53,11 +57,27 @@ func (d *Default) Apply(cluster *v2.Cluster) error {
 		return err
 	}
 	clusterRootfs := common.DefaultTheClusterRootfsDir(cluster.Name)
+
+	ex := shell.NewLex('\\')
+	var buildArgs []string
+	if image.Spec.ImageConfig.Args != nil {
+		buildArgs = append(buildArgs, utils.ConvertMapToEnvList(image.Spec.ImageConfig.Args)...)
+	}
+	if len(cluster.Spec.Env) != 0 {
+		buildArgs = append(buildArgs, cluster.Spec.Env...)
+	}
+	arg := utils.ConvertEnvListToMap(buildArgs)
 	for i := range image.Spec.Layers {
 		if image.Spec.Layers[i].Type != common.CMDCOMMAND {
 			continue
 		}
-		if err := sshClient.CmdAsync(runtime.GetMaster0Ip(cluster), fmt.Sprintf(common.CdAndExecCmd, clusterRootfs, image.Spec.Layers[i].Value)); err != nil {
+
+		cmdline, err := ex.ProcessWordWithMap(image.Spec.Layers[i].Value, arg)
+		if err != nil {
+			return fmt.Errorf("failed to render build args: %v", err)
+		}
+
+		if err := sshClient.CmdAsync(runtime.GetMaster0Ip(cluster), fmt.Sprintf(common.CdAndExecCmd, clusterRootfs, cmdline)); err != nil {
```

Comment:
`ex.ProcessWordWithMap(image.Spec.Layers[i].Value, arg)`
I think this `arg` should be cluster.env and support passing in parameters when sealer apply

Reply:
yes, "arg" is the merged parameter from cluster.env and image  build-arg, and i already do some overwrite action.